### PR TITLE
Include __all__ iff the implementation has it

### PR DESCRIPTION
**Pull request by srittau**
Closes: #67

### CODE REVIEW

File: pep-9999.rst [hunk 1]
```diff
@@ -357,7 +357,10 @@ specified in ``__all__`` are imported::
   public_attr: int
   _private_looking_public_attr: int
   private_attr: int
-  
+
+A type stub should contain an ``__all__`` variable if and only if the
+implementation has one too, and it should be identical to the implementation.
```
Comment (JelleZijlstra) [inline pep-9999.rst]: Something to consider: what if `__all__` is dynamically generated at runtime? We should probably say that it should be equal to the actual runtime value, even if at runtime it uses concatenation or manual list manipulation to construct `__all__`.

Comment (srittau) [reply pep-9999.rst]: Good point, I will change the wording to that effect. I also move this paragraph to the "What to include" section, where we also already have a paragraph about `__all__`.

Comment (srittau) [reply pep-9999.rst]: Update: It existed in the style guide section. I am shuffling all of this around a bit.

**Comment by srittau** [comment]
The PR now does the following:

* Straight up disallow tuples and sets.
* Disallow dynamic modification.
* This makes the style guide section superfluous.
* Add section in "What to include".

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**